docs/configuration.md

```diff
@@ -154,6 +154,74 @@ queries or when latency is particularly important.
 
 Defaults to `False`.
 
+## `GUARDIAN_CACHE_ANONYMOUS_USER`
+
+!!! abstract "Added in version 3.1.2"
+
+When set to `True`, the `get_anonymous_user()` function will cache the
+anonymous user instance to avoid repetitive database queries. Since the
+anonymous user configuration (`ANONYMOUS_USER_NAME`) is set at application
+startup and doesn't change during runtime, caching is safe and can provide
+significant performance improvements in applications that frequently access
+the anonymous user.
+
+When set to `False` (default), each call to `get_anonymous_user()` will
+perform a fresh database query.
+
+!!! tip "Performance optimization"
+
+    If your application frequently calls `get_anonymous_user()` or uses
+    object permissions with anonymous users, enabling this setting can
+    reduce database load and improve response times.
+
+!!! warning "Cache persistence"
+
+    The cache persists for the lifetime of the Python process. If you
+    manually change the anonymous user in the database during runtime
+    (which is not recommended), you'll need to restart your application
+    for the changes to take effect when caching is enabled.
+
+Defaults to `False`.
+
+## `GUARDIAN_ANON_CACHE_TTL`
+
+!!! abstract "Added in version 3.1.3"
+
+Sets the cache timeout (in seconds) for the anonymous user when
+`GUARDIAN_CACHE_ANONYMOUS_USER` is enabled. This determines how long
+the anonymous user instance will remain in the cache before being
+refreshed from the database.
+
+This setting uses Django's cache framework, making it compatible with
+all Django cache backends (Redis, Memcached, database cache, etc.)
+and supporting distributed deployments.
+
+```python
+# Cache anonymous user for 10 minutes
+GUARDIAN_CACHE_ANONYMOUS_USER = True
+GUARDIAN_ANON_CACHE_TTL = 600
+
+# Cache anonymous user for 1 hour
+GUARDIAN_ANON_CACHE_TTL = 3600
+
+# Cache anonymous user indefinitely (not recommended)
+GUARDIAN_ANON_CACHE_TTL = None
+```
+
+!!! tip "Performance tuning"
+
+    - For high-traffic applications: Use shorter TTL (300-600 seconds)
+    - For stable applications: Use longer TTL (1800-3600 seconds)
+    - Consider your cache backend's memory limits when setting TTL
```

Comment:
I like this advice, it always helps to have a bit of experience showing up in docs like this. Educate me @Natgho - why would you use a shorter TTL for high traffic apps?